fix rapidxml to compile with gcc

carmichaelong opened the pull request:
This PR adds forward declarations necessary for GCC 4.7 and above. Also adds a missing #include. @tgeijten

tgeijten commented:
Thanks for the update and sorry for the slow response! W.R.T. rapidxml, is it possible to take the forward declarations out of `rapidxml_print.hpp` and move them right before when `rapidxml_print.hpp` is included? I would like to keep 3rd party libs clean in case of updates.

W.R.T. flut: the include you added shouldn't be needed in the current version, can you verify this?

carmichaelong commented:
I've updated the PR to move the forward declaration out and keep rapidxml clean.

I found that I still needed the include (for `mid_str`), so that is still in this PR. I updated it with some other fixes that are needed for GCC with some of the recent commits.

tgeijten commented on flut/buffer/storage.hpp:
```diff
@@ -23,7 +23,7 @@ namespace flut
 
 		/// find index of a label
-		index_t find_channel( const L& label ) const { auto it = label_indices_.find( label ); return ( it != label_indices_.end() ) ? it->second : no_index; }
+		index_t find_channel( const L& label ) const { auto it = labels_.find( label ); return ( it != labels_.end() ) ? it->second : no_index; }
```
Just noticed this after merging: it doesn't seem right, could you clarify?

tgeijten commented:
I fixed this on the master branch, can you check if it still compiles with gcc?

carmichaelong commented:
This issue looks fixed on gcc, though gcc will not compile right now due to new code that was added. I will put in a new issue about this.
